vllm/worker/openvino_model_runner.py

```diff
@@ -0,0 +1,340 @@
+from typing import List, NamedTuple, Optional, Tuple
+
+import openvino as ov
+import torch
+from torch import nn
+
+from vllm.attention import get_attn_backend
+from vllm.attention.backends.openvino import OpenVINOAttentionMetadata
+from vllm.config import (CacheConfig, DeviceConfig, LoadConfig, LoRAConfig,
+                         ModelConfig, ParallelConfig, SchedulerConfig,
+                         VisionLanguageConfig)
+from vllm.logger import init_logger
+from vllm.model_executor import SamplingMetadata
+from vllm.model_executor.model_loader.openvino import get_model
+from vllm.sequence import SamplerOutput, SequenceGroupMetadata
+
+logger = init_logger(__name__)
+
+
+class ModelInput(NamedTuple):
+    input_tokens: torch.Tensor
+    input_positions: torch.Tensor
+    attn_metadata: Optional[OpenVINOAttentionMetadata]
+    seq_lens: List[int]
+    query_lens: List[int]
+    multi_modal_input: Optional[torch.Tensor]
+
+    @classmethod
+    def empty(cls, device):
+        return ModelInput(input_tokens=torch.empty(0, device=device),
+                          input_positions=torch.empty(0, device=device),
+                          attn_metadata=None,
+                          seq_lens=[],
+                          query_lens=[],
+                          multi_modal_input=None)
+
+
+class OpenVINOModelRunner:
```

Comment:
Can you add some tests for the ModelRunner?

Reply:
Can we implement as next step in our RFC for OpenVINO integration?

BTW, I don't see any tests for any model runners except CUDA ones. How it supposed to run such tests? Does vLLM provide an ability to build vLLM with specific backend on CI service and run backend tests? So far, I have seen small scripts like `run-cpu-test.sh`, which can ensure e2e flow works and integration is not broken.

Reply:
BTW, I don't see that scripts `run-xpu-test.sh`, `run-cpu-test.sh` are currently used, while previously they were mentioned in some CI scripts. How can we ensure e2e on simple model is run on CI?

Reply:
This makes sense for a followup PR. Maybe @simon-mo can clarify how to go about new CI backends?